build_tools/doctool/src/com/google/doctool/custom/JavaEmulSummaryDoclet.java

```diff
@@ -16,281 +16,484 @@
 
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.PackageElement;
+import javax.lang.model.type.ArrayType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.type.TypeVariable;
+import javax.tools.Diagnostic;
+
+import jdk.javadoc.doclet.Doclet;
+import jdk.javadoc.doclet.DocletEnvironment;
+import jdk.javadoc.doclet.Reporter;
+
 /**
  * A doclet for listing the specified classes and
  * their methods and constructors.
  */
 public class JavaEmulSummaryDoclet implements Doclet {
 
-    public static final String OPT_OUTFILE = "-outfile";
-    private static final String JAVADOC_URL = "https://docs.oracle.com/en/java/javase/11/docs/api/";
-
-    private Reporter reporter;
-    private String outputFile;
-
-    @Override
-    public boolean run(DocletEnvironment env) {
+  public static final String OPT_OUTFILE = "-outfile";
+  // should be aligned with the latest version that's tested by CI
+  private static final int MAX_JRE_VERSION = 22;
+  private static final int CURRENT_JRE_VERSION = Runtime.version().feature();
+  private static final String JAVADOC_URL = "https://docs.oracle.com/en/java/javase/"
+      + CURRENT_JRE_VERSION + "/docs/api/";
+  private static final List<String> DEPRECATED = List.of(
+      "java.lang.Character#isJavaLetter(char)",
+      "java.lang.Character#isJavaLetterOrDigit(char)",
+      "java.lang.String#getBytes(int, int, byte[], int)"
+  );
+
+  private Reporter reporter;
+  private String outputFile;
+
+  private final Map<String, String> ghToSignatures = new HashMap<>();
+  private final Map<String, String> ghToTitle = new HashMap<>();
+  private final List<String> triage = new ArrayList<>();
+
+  @Override
+  public boolean run(DocletEnvironment env) {
+    final Properties properties = new Properties();
+    Path missingDir = Path.of("build_tools/doctool/src/com/google/doctool/custom/missing");
+    try (Stream<Path> str = Files.list(missingDir)) {
+      str.filter(file -> !file.getFileName().toString().startsWith("triage")).forEach(file -> {
         try {
+          properties.clear();
+          properties.load(Files.newInputStream(file));
+          String issue = file.getFileName().toString().replaceAll("gh|\\.properties", "");
+          ghToSignatures.put(issue, (String) properties.get("members"));
+          ghToTitle.put(issue, (String) properties.get("title"));
+        } catch (IOException ex) {
+          throw new UncheckedIOException(ex);
+        }
+      });
+    } catch (IOException ex) {
+      throw new UncheckedIOException(ex);
+    }
+    File outFilePresent = new File(outputFile);
+    File outFileMissing = new File(outputFile.replace(".html", "-missing.html"));
+    outFilePresent.getParentFile().mkdirs();
```

Comment:
Files+Path instead here? Better errors for mkdirs() than just ignoring it if we can't write.